src/solvers/mp/rcspp/RCGraph.cpp

```diff
@@ -0,0 +1,1095 @@
+      bool b_is_a_path_at_all = false;
+      bool b_feasible = false;
+      bool b_correctly_extended = false;
+      spp_res_cont actual_final_resource_levels( 0, initial_label_values );
+      boost::graph_traits<Graph>::edge_descriptor ed_last_extended_arc;
+      check_r_c_path( g_,
+                      opt_solutions_spp[p],
+                      spp_res_cont( 0, initial_label_values ),
+                      true,
+                      pareto_opt_rcs_spp[p],
+                      actual_final_resource_levels,
+                      ref_spp(),
+                      b_is_a_path_at_all,
+                      b_feasible,
+                      b_correctly_extended,
+                      ed_last_extended_arc );
+      // b. if feasible, add the solution
+      if( b_is_a_path_at_all && b_feasible && b_correctly_extended )
+      {
+        if(pareto_opt_rcs_spp[p].cost < maxReducedCostBound)
+          rc_solutions.push_back(solution(opt_solutions_spp[p], pareto_opt_rcs_spp[p]));
+      }
+      // c. print a warning as it shouldn't be the case
+      else {
+        if( !b_is_a_path_at_all )
+          std::cerr << "Not a path." << std::endl;
+        if( !b_feasible )
+          std::cerr << "Not a feasible path." << std::endl;
+        if( !b_correctly_extended )
+          std::cerr << "Not correctly extended." << std::endl;
+      }
+
+    }
+
+    return rc_solutions;
+}
+
+RCSolution RCGraph::solution(
+    const std::vector< boost::graph_traits<Graph>::edge_descriptor >& path,
+    const spp_res_cont& resource){
+#ifdef DBG
+//  printPath(path, resource);
+#endif
+
+  RCSolution sol(resource.cost);
+
+  // All arcs are consecutively considered
+  //
+  for( int j = static_cast<int>( path.size() ) - 1; j >= 0;	--j){
+    int a = boost::get(&Arc_Properties::num, g_, path[j]);
+    int day = arcDay(a);
+    const std::vector<int>& shifts = arcShifts(a);
+
+    if(day != -1 && !shifts.empty()) {
+      // if it's the first day
+      if(sol.firstDay == -1) sol.firstDay = day;
+      // append the shifts
+      for(int s: shifts) sol.shifts.push_back(s);
+    }
+  }
+
+  return sol;
+}
+
+// Addition of a single node
+int RCGraph::addSingleNode(NodeType type, std::vector<int> lbs, std::vector<int> ubs){
+  add_vertex( Vertex_Properties( nNodes_, type, lbs, ubs ), g_ );
+  return nNodes_++;
+}
+
+// Adds a single arc (origin, destination, cost, travel time, type)
+int RCGraph::addSingleArc(int o, int d, double baseCost, std::vector<int> consumptions,
+    ArcType type, int day, std::vector<int> shifts){
+  boost::graph_traits< Graph>::edge_descriptor e =
+      (add_edge( o, d, Arc_Properties( nArcs_, type, baseCost, consumptions, day, shifts ), g_ )).first;
+  arcsDescriptors_.push_back(e);
+  return nArcs_++;
+
+//  if(nodeType(o) == PRINCIPAL_NETWORK
+//     and nodeType(d) == PRINCIPAL_NETWORK
+//     and principalToDay_[o] < principalToDay_[d]-1){
+//    printArc(nArcs_-1);
+//    //		getchar();
+//  }
```

Comment:
I guess we can delete this comment. It must belong to an old debug session of Sam

Reply:
done in 7e2095d7454a9abd9fd2e1b6bd6ba14634d42a66